### DISCUSSION TITLE
What are the changes with Draft mode?

### DISCUSSION THREAD

**Discussion by angry-dan** [vercel/next.js#49613] [Help]
### Summary

[Preview mode has been deprecated](https://nextjs.org/docs/pages/building-your-application/configuring/preview-mode). Draft mode replaces it. Except I can't seem to find any documentation explaining why the API needed to change, or indeed what changes are being made. Can anyone say what the differences are, what the upgrade path is, and why?

**Comment by aaamenezes** [comment]
I would like to know too. Incidentally, the new format didn't work for me:

- Removed the async word
- I changed setPreviewData for setDraftMode

But I get the error `res.setDraftMode is not a function`.

Not clear if it only works on a specific version of Next.JS

**Reply by MauriceArikoglu** [reply]
Same issue here @aaamenezes

**Comment by andrewfairlie** [comment]
Also interested in this.

It also seems to have a big shortcoming because our CMS provides a "preview token" for each revision of the article which it provides as a URL param. Previously we used `setPreviewData` to read/write this and add that as appropriate to our API calls but this is no longer an option as far as I can tell.

**Reply by jams2** [reply]
Same issue for my team — we pass a preview token through using `setPreviewData` to enable previewing pages from our headless CMS.

**Comment by noisehunter82** [comment]
We updated to Next 13 and it looks like DraftMode doesn't allow passing any custom data to the page route component. We rely on this feature, as it allows us to preview dynamic routes by pulling page data from Conentful by entry ID in selected environments. Our app is used to build several sites. Passing custom data to page routes let's us a single app to render previews of pages for various sites. How can I make Draft Mode work with dynamic routes and pass request data to my dynamic page routes?

**Comment by jwickens** [comment]
We need `previewData` for E2E testing as shown in this example repo: https://github.com/jwickens/next-page-e2e-testing/tree/main

**Comment by harry-gocity** [comment]
Did anyone ever solve the removal of `previewData`?
